path_login.go

```diff
@@ -0,0 +1,229 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/pkg/errors"
+	"net/http"
+	"strings"
+	"unicode"
+)
+
+// These constants store the required http path & method information for validating the signed request
+const (
+	PathVersionBase = "/v1"
+	PathBaseFormat  = "/auth/oci/login/%s"
+	PathLoginMethod = "get"
+)
+
+// Signing Header constants
+const (
+	// HdrRequestTarget represents the special header name used to refer to the HTTP verb and URI in the signature.
+	HdrRequestTarget = `(request-target)`
+)
+
+func pathLogin(b *backend) *framework.Path {
+	return &framework.Path{
+		Pattern: "login/" + framework.GenericNameRegex("role"),
+		Fields: map[string]*framework.FieldSchema{
+			"requestHeaders": {
```

Comment:
This should be `request_headers` for consistency.

Reply:
Done